From: Harsh Jain <harsh@chelsio.com>
Date: Tue, 6 Mar 2018 10:37:51 +0530
Subject: crypto: chelsio - Fix iv passed in fallback path for rfc3686
Patch-mainline: v4.17-rc1
Git-commit: 7ffb911882a34a0d1f05ba64d78bf9b61d1e4179
References: bsc#1097585 bsc#1097586 bsc#1097587 bsc#1097588 bsc#1097583 bsc#1097584

We use ctr(aes) to fallback rfc3686(ctr) request. Send updated IV to fallback path.

Signed-off-by: Harsh Jain <harsh@chelsio.com>
Signed-off-by: Herbert Xu <herbert@gondor.apana.org.au>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/crypto/chelsio/chcr_algo.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/crypto/chelsio/chcr_algo.c
+++ b/drivers/crypto/chelsio/chcr_algo.c
@@ -1284,7 +1284,7 @@ static int process_cipher(struct ablkcip
 					   req->src,
 					   req->dst,
 					   req->nbytes,
-					   req->info,
+					   reqctx->iv,
 					   op_type);
 		goto error;
 	}
